go-controller/pkg/ovn/policy.go

```diff
@@ -300,16 +301,28 @@ func (oc *DefaultNetworkController) syncNetworkPolicies(networkPolicies []interf
 	}
 
 	stalePGs := []string{}
-	err := oc.addressSetFactory.ProcessEachAddressSet(func(addrSetName, namespaceName, policyName string) error {
-		if policyName != "" && !expectedPolicies[namespaceName][policyName] {
-			// policy doesn't exist on k8s. Delete the port group
-			portGroupName := fmt.Sprintf("%s_%s", namespaceName, policyName)
-			hashedLocalPortGroup := hashedPortGroup(portGroupName)
-			stalePGs = append(stalePGs, hashedLocalPortGroup)
-			// delete the address sets for this old policy from OVN
-			if err := oc.addressSetFactory.DestroyAddressSetInBackingStore(addrSetName); err != nil {
-				klog.Errorf(err.Error())
-				return err
+	err := oc.addressSetFactory.ProcessEachAddressSet(func(hashedName, addrSetName string) error {
+		// netpol name has format fmt.Sprintf("%s.%s.%s.%d", gp.policyNamespace, gp.policyName, direction, gp.idx)
```

Comment:
the format is really <namespace>.<policy>.<direction>.<index>_<v4/v6>, but I guess that doesn't matter for ProcessEachAddressSet will truncate the _v4/v6. Might be good to add a comment here stating that

Reply:
that _<v4/v6> part is (should) only be known to the address set factory, it adds that suffix and it crops it in `ProcessEachAddressSet`, so that callers don't need to think about it.
So I think it makes sense to leave that knowledge to the factory, wdyt?